crates/node-api/src/evm/traits.rs

```diff
@@ -0,0 +1,49 @@
+use reth_primitives::{revm::env::fill_block_env, Address, ChainSpec, Header, Transaction, U256};
+use revm::{Database, Evm, EvmBuilder};
+use revm_primitives::{BlockEnv, CfgEnvWithHandlerCfg, SpecId, TxEnv};
+
+/// Trait for configuring the EVM for executing full blocks.
+pub trait ConfigureEvm: ConfigureEvmEnv {
+    /// Returns new EVM with the given database
+    fn evm<'a, DB: Database + 'a>(&self, db: DB) -> Evm<'a, (), DB> {
+        EvmBuilder::default().with_db(db).build()
+    }
+
+    /// Returns a new EVM with the given inspector
+    fn evm_with_inspector<'a, DB: Database + 'a, I>(&self, db: DB, inspector: I) -> Evm<'a, I, DB> {
+        EvmBuilder::default().with_db(db).with_external_context(inspector).build()
+    }
+}
+
+/// This represents the set of methods used to configure the EVM before execution.
+pub trait ConfigureEvmEnv: Send + Sync + Unpin + Clone {
+    /// The type of the transaction metadata.
+    type TxMeta;
```

Comment:
nit: add docs on what transaction metadata is in this context (e.g. give an example)